requests/adapters.py

```diff
@@ -150,14 +156,26 @@ def proxy_manager_for(self, proxy, **proxy_kwargs):
         :returns: ProxyManager
         """
         if not proxy in self.proxy_manager:
```

Comment:
While we're here, let's reformat this a bit:

```py
if proxy in self.proxy_manager:
    manager = self.proxy_manager[proxy]
elif proxy.lower().startswith('socks'):
    username, password = get_auth_from_url(proxy)
    manager = self.proxy_manager[proxy] = SOCKSProxyManager(
        # ...
    )
else:
    proxy_headers = self.proxy_headers(proxy)
    manager = self.proxy_manager[proxy] = proxy_from_url(
        # ...
    )
return manager
```

I feel like this will be easier to read